lib/src/reactive_widget.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:io';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+abstract class ReactiveWidget<T> extends StatefulWidget {
+  final Stream<T> stream;
+  final T initialData;
+  Widget build(BuildContext context, T data);
+
+  @mustCallSuper
+  const ReactiveWidget(this.stream, this.initialData, {Key key})
+      : assert(stream != null),
+        super(key: key);
+
+  Widget errorBuild(BuildContext context, Object error) {
+    return Center(
+      child: Text(error),
+    );
+  }
+
+  Widget streamEmptyBuild(BuildContext context) {
```

Comment:
Wouldn't the name BusyBuilder here be better than empty builder?

Reply:
I didn't think of anything better when I put this name. And the name is really not very cool.
Looking at RxComandBuilder and RxLoader, when the stream has no data and no error the return is placeHolderBuilder.

```Dart
if (!lastReceivedItem.hasData && !lastReceivedItem.hasError) {
      if (widget.placeHolderBuilder != null) {
        return widget.placeHolderBuilder(context);
      }
    }
```

Since the behavior is the same, wouldn't it be better to rename it to placeHolderBuilder?

Reply:
Sorry I somehow thought I had seen it used in case of a waiting StreamBuilder. I think `placeHolderBuilder` is a better solution. I saw that you named them `Build`and not `Builder`any reason for that?

Reply:
I put it as 'Build' instead of 'Builder' to follow the 'build' method.
And also to differentiate when it is the internal method or when it receives as a parameter.
When it is received as a parameter.
What do you think about ? Is it good that way?

``` dart
// when it comes to the internal method.
Widget build(BuildContext context, T data);
Widget errorBuild(BuildContext context, Object error);
Widget placeHolderBuild(BuildContext context);
``` 
``` dart
// When it is received as a parameter.
final RxBuilder<T> builder;
final ErrorBuilder<T> errorBuilder;
final PlaceHolderBuilder placeHolderBuilder;
```